src/component-queries/convert-data.ts

```diff
@@ -3,33 +3,47 @@ import {
     NormalCellLineNode,
     UnpackedDiseaseCellLine,
     UnpackedGene,
-    GeneFrontMatter,
     UnpackedNormalCellLine,
+    GeneticModification,
 } from "./types";
 
+const extractGeneticModifications = (
+    modifications?: GeneticModification[]
+): { taggedGene: UnpackedGene, alleleCount: string, tagLocation: string, fluorescentTag: string }[] => {
+    if (!modifications || !modifications.length) return [];
 
-const extractGenes = (geneArray: { frontmatter: GeneFrontMatter }[] = []): UnpackedGene[] => {
-    if (!geneArray) return [];
-
-    return geneArray
-        .filter((gene) => gene && gene.frontmatter) 
-        .map((gene) => ({
-            name: gene.frontmatter.name,
-            symbol: gene.frontmatter.symbol,
-            structure: gene.frontmatter.structure,
-            protein: gene.frontmatter.protein,
-    }));
+    return modifications
+        .filter(mod => mod && mod.gene && mod.gene.frontmatter)
+        .map(mod => ({
+            taggedGene: {
+                name: mod.gene?.frontmatter.name || "",
+                symbol: mod.gene?.frontmatter.symbol || "",
+                structure: mod.gene?.frontmatter.structure || "",
+                protein: mod.gene?.frontmatter.protein || ""
+            },
+            alleleCount: mod.allele_count || "",
+            tagLocation: mod.tag_location || "",
+            fluorescentTag: mod.fluorescent_tag || ""
+        }));
```

Comment:
I think it makes more sense to extract these into 3 separate arrays. it'll make using them in the table much simpler. So that way you can still have the column name be alleleCount, but it just receives an array, and same for tagLocation and fluorescentTag